prepare.sh

```diff
@@ -1,7 +1,72 @@
 #!/bin/sh
 
+ARGUMENT_LIST=(
+    "ref"
+    "workdir"
+    "gitrepo"
+)
+
+opts=$(getopt \
+    --longoptions "$(printf "%s:," "${ARGUMENT_LIST[@]}")help" \
+    --name "$(basename "${0}")" \
+    --options "" \
+    -- "$@"
+)
+
+eval set -- ${opts}
+
+while true; do
+    case "${1}" in
+    --help)  
+        shift
+        echo "Options:"
+        echo "--help|-h                 specify the flags"
+        echo "--ref                     specify the reference of pr"
+        echo "--workdir                 specify the working directory"
+        echo "--gitrepo                 specify the git repository" 
+        echo " "
+        echo "Sample Usage:"
+        echo "./prepare.sh --gitrepo=https://github.com/example --workdir=/opt/build --ref=pull/123/head"
+        exit 0
+        ;;
+    --gitrepo)  
+        shift
+        gitrepo=${1}
+        ;;
+    --workdir)  
+        shift
+        workdir=${1}
+        ;;
+    --ref)  
+        shift
+        ref=${1}
+        echo ${ref}      
+        ;;
+    --)
+        shift
+        break
+        ;;
+    esac
+    shift
+done
+
 set -x
 
 yum -y install git podman
 
-git clone --single-branch --branch=master https://github.com/ceph/ceph-csi /opt/build
+# These values can be modified as per the requirement.
+defaultRepo="https://github.com/ceph/ceph-csi"
+defaultWorkDir="tip/"
+defaultRef="master"
+
+# The variables will be set corresponding to passed cli parameters.
+# In case no value is specified, default values will be used.
+gitrepo=${gitrepo:-"$(echo ${defaultRepo})"}
+workdir=${workdir:-"$(echo ${defaultWorkDir})"}
+ref=${ref:-"$(echo ${defaultRef})"}
+
+
+git clone --depth=1 ${gitrepo} ${workdir}
+cd ${workdir}
+git fetch origin "${ref}:tip/${ref}"
```

Comment:
consider calling `git fetch --depth=1 origin ...` here, it will prevent downloading the full history of the ref and things will consume less bandwidth and go a bit faster.